drivers/espi/espi_emul.c

```diff
@@ -203,28 +521,36 @@ int espi_emul_register(const struct device *dev, const char *name,
 {
 	struct espi_emul_data *data = dev->data;
 
-	sys_slist_append(&data->emuls, &emul->node);
+	__ASSERT_NO_MSG(data->emul == NULL);
+	data->emul = emul;
 
 	LOG_INF("Register emulator '%s' at cs %u\n", name, emul->chipsel);
 
 	return 0;
 }
 
 /* Device instantiation */
-static struct emul_espi_driver_api emul_espi_driver_api = {
+static struct espi_emul_driver_api api = {
 	.espi_api = {
 		.config = espi_emul_config,
 		.get_channel_status = espi_emul_get_channel_status,
-		.read_lpc_request = espi_emul_read_lpc_request,
-		.write_lpc_request = espi_emul_write_lpc_request,
+		.read_request = espi_emul_read_request,
+		.write_request = espi_emul_write_request,
+		.read_lpc_request = espi_emul_lpc_read_request,
+		.write_lpc_request = espi_emul_lpc_write_request,
 		.send_vwire = espi_emul_send_vwire,
 		.receive_vwire = espi_emul_receive_vwire,
-		.manage_callback = espi_emul_manage_callback
+		.send_oob = espi_emul_send_oob,
+		.receive_oob = espi_emul_receive_oob,
+		.flash_read = espi_emul_flash_read,
+		.flash_write = espi_emul_flash_write,
+		.flash_erase = espi_emul_flash_erase,
+		.manage_callback = espi_emul_manage_callback,
```

Comment:
It looks like some of there don't have the `static` keyword, do they need to have external linkage?

Reply:
Only some of them needs to be visible, fixed